csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -885,6 +885,16 @@ internal static string GetOriginalName(object value)
                 return originalName;
             }
 
+#if DOTNET35
+            private static Dictionary<object, string> GetNameMapping(System.Type enumType) =>
```

Comment:
Could you add a TODO here:

// TODO: Consider adding functionality to TypeExtensions to avoid this difference.

It would be nice to keep as many differences as possible within the Compatibility namespace.